### DISCUSSION TITLE
Future of web resources

### DISCUSSION THREAD

**Discussion by jlillis** [multitheftauto/mtasa-resources#279] [Ideas]
We currently have two documented issues with our web resources (#278 and #130). While these issues are maybe not so hard to resolve, there has been some discussion on Discord about rewriting or deprecating the web resources. I'm opening this discussion to organize the conversation.

In particular, there are two questions I think we should start with:
1. Should we just deprecate the web resources entirely?
2. If not, then what improvements should be made?

**Comment by jlillis** [comment]
My personal opinion is that:
1. Deprecating would be fine, since it does not seem to me that the web resources are particular useful or popular. I'm sure at least some may disagree.
2. If we are going to make any improvements, we might as well go all the way and implement visual and functional improvements to the existing web resources. I do not think a complete rewrite would be a good idea as it may or may not be finished in a timely manner. We should try to aim for project completion before the next major release of MTA (or, at the very least, fix any existing bugs we can find).

**Comment by Dante383** [comment]
Here's my opinion: 

1. We should deprecate the current web resources. They're archaic, not really updated. 
2. We should write new web resources from scratch. Use moderns technologies, like Bootstrap for CSS and Vuejs for, well, JS. Improve feature list - in 10 years since the last major upgrades in web resources MTA has added a lots of functions that can be used.

**Reply by Dante383** [reply]
After giving it some thought, maybe not vuejs - that would raise the entry-level for creating own features a little bit too much. Just plain JS

**Reply by Dante383** [reply]
About the structure - we should create one main resource handling the webpage and allow other resources to add it's own tabs through exported functions

**Reply by qaisjp** [reply]
> About the structure - we should create one main resource handling the webpage and allow other resources to add it's own tabs through exported functions

Doesn't something already do this?

- iirc there's a concept of a "default resource" (`:22005/`), which is by default configured to resourcemanager or webadmin or something, I don't remebmer
- every other resource is accessed under the resource name (`:22005/runcode`)
- but everything is actually accessed via the tabs in that default resource (which currently iframes to each subdirectory/resource)

**Reply by Dante383** [reply]
Yeah, that's how it works now. We were however thinking about rewriting current resources from scratch so that's why I am mentioning that.

**Comment by patrikjuvonen** [comment]
I'll look into the existing web resources and see what could be the next steps.

**Comment by Disinterpreter** [comment]
My opinion is:
1. Should we just deprecate the web resources entirely?
Yes, we can, but for the first step, we can try to change something. I mean the current [web] must be deleted, but we can write new, simpler, closer to newbie etc...
2. If not, then what improvements should be made?

Take a look on the resources list, I crossed out useless and deprecated for my opinion.
- ajax (?????)
- ~elementbrowser~
- performancebrowser (rewriting for new webadmin)
- ~resourcebrowser~ 
- ~resourcemanager~ 
- webadmin (full rewriting)
- ~webmap~ 
- ~webstats~

Moreover, this resources was created more than 10 years ago, when websites were more convenient than today.
I mean, if we want to do some convenient new thing, we need to rewrite it and probably trying to add some bot API support (Discord, Telegram etc...) because today messengers more popular and convenient for people.

P.S I disagree to delete this resources without alternative solution.

**Reply by Dante383** [reply]
I believe ajax is used to dynamically change the website, like update player's ping without refreshing the page

**Reply by patrikjuvonen** [reply]
I think they're all useful and remind people that these features (http functions, scripting language) are built in MTA. I used to use them a lot back in the days and I still would if I ran a server actively. Let's just refresh them up, at least fix all the errors and make them work. A style revamp etc. can be done as a separate project.

**Reply by Dante383** [reply]
I think it'd be way easier (and quicker) to just rewrite them from scratch - if we wanted to fix it we would've to rewrite 90% of the resource anyway

**Comment by jlillis** [comment]
@patrikjuvonen has opened #281 which not only resolves #130 but makes a number of additional improvements. In light of this, I think it would be best to make quick fixes and improvements to the web resources now, so we know we'll have them working for the next MTA release. There's only one documented issue left (#278), but it would be good to test for others. We can then continue this discussion to determine what further changes we'd like to make.

**Comment by jlillis** [comment]
With #293 merged into master, there are no longer any known issues with the web resources. If nobody else has any immediate/short-term plans for further improvements (or issues to report), I'd like to consider this matter resolved for now.
